### ISSUE TITLE
setNewestActiveCart func in cartParse.js isn't properly sorting by date

### ISSUE THREAD

**Issue by ismurray** [RaindropAndWee/raindrop-client#9]
In order to account for the possibility that through either malicious injection via curl script, or through errors during testing, there may be more than one 'active' cart on a given account, I tried to write a function that would consistently choose "the one true cart" based on some consistent factor (I chose most recent createdAt date).

The function is supposed to filter all carts where purchased === false (i.e. 'active' carts) into an array. Then, it sorts this array by date, and sets the most recently created cart to store.activeCart, essentially singling that cart out for all cart actions like add item, remove item, etc.

For some reason it seems to always be choosing the oldest cart instead. Even when I reverse the sort parameters. I'm not sure why this is. It shouldn't matter whether it always chooses the oldest or the newest, as long as it's predictable in which cart it chooses. More testing needs to be done to figure out what is going wrong.

This could be made completely unnecessary if we figure out some form of validation on the back end that only allows one 'active' cart at any given time.

**Comment by ismurray** [comment]
The function in question: 
```js
// If there are more than one active carts (due to some internal error or
// malicious injection), choose the newest one
const setNewestActiveCart = function () {
  const activeCarts = store.allMyCarts.filter((cart) => cart.purchased === false)
  // sorts all the active carts by date
  activeCarts.sort(function (a, b) {
    return b.createdAt - a.createdAt
  })
  console.log('sorted activeCarts are ', activeCarts)
  store.activeCart = activeCarts[0]
  console.log('activeCart is ', store.activeCart)
}
```

**Comment by ismurray** [comment]
Resolved in commit eb54efb
